src/control/server/telemetry.go

```diff
@@ -80,16 +92,14 @@ func startPrometheusExporter(ctx context.Context, log logging.Logger, port int,
 	listenAddress := fmt.Sprintf("0.0.0.0:%d", port)
 	log.Infof("Listening on %s", listenAddress)
 
-	if err := http.ListenAndServe(listenAddress, nil); err != nil {
-		return err
-	}
-
-	return nil
-}
+	// http listener is a blocking call
+	go func() {
+		err := http.ListenAndServe(listenAddress, nil)
+		log.Infof("Prometheus web exporter stopped: %s", err.Error())
+	}()
 
-func stopPrometheusExporter() {
-	for idx, cleanupSrc := range cleanupTelem {
-		cleanupSrc()
-		delete(cleanupTelem, idx)
-	}
+	return func() {
+		log.Debug("cleaning up telemetry bindings")
+		cleanupAll(cleanupFns)
```

Comment:
nit: Might be clearer to just iterate over the slice here in the closure, otherwise the reader has to go look at the implementation of `cleanupAll()` to figure out what's going on.

Reply:
Good point. I can fix in a follow on.